### PR TITLE
Support Azure Blob Storage

### PR DESCRIPTION
**Pull request by yp05327**
Try to support #22527

Using azure official SDK, add a new storage type: azureblob

This PR is working in process.

TODO:
- [ ] CI Tests
  - [ ] integration test
  - [ ] unit test 
- [ ] Improve performance?
- [ ] CLI Migrate Storage
- [ ] docs?

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Why the MinIO SDK (which is a S3 client) doesn't work with Azure Blob Storage? Isn't Azure Blob Storage S3-compatible ?

**Comment by techknowlogick** [comment]
>  Isn't Azure Blob Storage S3-compatible ?

no, it uses a different API. https://learn.microsoft.com/en-us/rest/api/storageservices/blob-service-rest-api

File: modules/storage/azure.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	Offset int64
+}
+
+func (a *azureBlobObject) downloadStream(p []byte) (int, error) {
+	if a.BlobClient == nil {
+		return 0, fmt.Errorf("no blob client in azure blob object")
+	}
+	if a.Context == nil {
+		return 0, fmt.Errorf("no context in azure blob object")
+	}
```
Comment (KN4CK3R) [inline modules/storage/azure.go]: These are never true?

Comment (yp05327) [reply modules/storage/azure.go]: It is based on the caller, maybe we can remove this. 🤔

Comment (KN4CK3R) [reply modules/storage/azure.go]: But it's a private method, so there are only "internal" callers. And every callpath sets these attributes.

Comment (yp05327) [reply modules/storage/azure.go]: You are right. We don't need these codes here.

**Comment by yp05327** [comment]
Download from direct URL uses 1mins, but download from Gitea uses 7mins. 😕

**Comment by yp05327** [comment]
Customizing `Content-Disposition` header was removed in the latest azure SDK.
Trying to fix it in https://github.com/Azure/azure-sdk-for-go/pull/22024

**Comment by wxiaoguang** [comment]
Would something like https://github.com/beyondstorage/go-storage help? It also helps GCP #26963

**Comment by yp05327** [comment]
> Would something like https://github.com/beyondstorage/go-storage help? It also helps GCP #26963

It seems that no one is maintaining it now. Only the bot is alive. 🤔

**Comment by yp05327** [comment]
replaced by #30995

**Pull request closed without merging**